iseq.h

```diff
@@ -13,6 +13,7 @@
 #include "internal/gc.h"
 #include "shape.h"
 #include "vm_core.h"
+#include "prism/prism.h"
```

Comment:
Is there a way we can do this without adding this dependency?

Reply:
I think, as long as `rb_iseq_new_main_prism` takes a `pm_string_t` and `pm_options_t`, we need those types available.

Maybe we can hide the types behind a `void *` or a `VALUE`? Would that be better?

I looked into if we could move this to `prism_compile.c` instead of defining this in `iseq.c`, but `prism_compile.c` doesn't know about `iseq_alloc`.

Also, it looks like `iseq.c` already depends on `prism/prism.h` via `prism_compile.h`. 

Is the idea in reducing dependencies to decouple things, or is the worry this will make compile times longer since changes in prism will require a recompile on all things that use `iseq.h`?

Reply:
The smell is actually a bit of `common.mk`. Historically we've been trying to keep the dependencies to a minimum to make it easier to pull Prism out if the Ruby core team ever decides to do something like that. But I hink in this case, you're right, it's not really worth the `void *` casting.